src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -313,7 +313,17 @@ public sealed override bool Equals(object obj)
         {
             var t2 = obj as TypeSymbol;
             if ((object)t2 == null) return false;
-            return this.Equals(t2, TypeCompareKind.ConsiderEverything);
```

Comment:
>TypeCompareKind.ConsiderEverything [](start = 35, length = 34)

Consider using `TypeCompareKind.AllNullableIgnoreOptions` directly since we already have two `TypeSymbol` instances, rather than going through the comparer and conversions from `object` or `ISymbol`.